common/persistence/dataInterfaces.go

```diff
@@ -719,6 +719,26 @@ type (
 		Size int
 	}
 
+	// AppendRawHistoryNodesRequest is used to append a batch of raw history nodes
+	AppendRawHistoryNodesRequest struct {
+		// The shard to get history node data
+		ShardID int32
+		// true if this is the first append request to the branch
+		IsNewBranch bool
+		// the info for clean up data in background
+		Info string
+		// The branch to be appended
+		BranchToken []byte
+		// The batch of events to be appended. The first eventID will become the nodeID of this batch
+		History *commonpb.DataBlob
+		// TransactionID for events before these events. For events chaining
+		PrevTransactionID int64
+		// requested TransactionID for this write operation. For the same eventID, the node with larger TransactionID always wins
+		TransactionID int64
```

Comment:
nit: not sure if it worth creating a common struct for Append and AppendRaw and embed it in those two requests.

Reply:
Good point

Reply:
Cannot think of a good name and structure. May do it in another PR.